src/Compilers/Core/Desktop/CommandLine/CommonCommandLineParser.cs

```diff
@@ -578,10 +578,52 @@ public static IEnumerable<string> SplitCommandLineIntoArguments(string commandLi
                 }))
             .Select(arg => arg.Trim())                                                                  // Trim whitespace
             .TakeWhile(arg => (!removeHashComments || !arg.StartsWith("#", StringComparison.Ordinal)))  // If removeHashComments is true, skip all arguments after one that starts with '#'
-            .Select(arg => CondenseDoubledBackslashes(arg).Unquote())                                   // Remove quotes and handle backslashes.
+            .Select(arg => UnquoteAndUnescape(NormalizeBackslashes(arg)))                                 // Remove quotes and handle backslashes.
             .Where(arg => !string.IsNullOrEmpty(arg));                        							// Don't produce empty strings.
         }
 
+        private static string UnquoteAndUnescape(string v)
```

Comment:
No, the purpose of SplitCommandLineIntoArguments is to separate the line into multiple arguments on whitespaces. It needs to maintain the inQuotes state to ensure we do not break line while in a quoted text. It also needs to count slashes since odd number of slashes before a quote escaped that quote.

Once the line is split into arguments we need to remove quotes that are not escaped, and need to remove slashes that are used for escaping. That is what UnquoteAndUnescape does.

NormalizeBackslashes is just to be able to deal with slashes once quotes are removed. Whether slashes are as-is or escaping depend on whether they precede a quote. Normalizing turns the input into a form that is not dependent on the presence of quotes. Thus I no longer have circular dependency and can remove quotes first and then remove escaping slashes.